src/otx/v2/adapters/torch/mmengine/mmpretrain/engine.py

```diff
@@ -38,6 +39,41 @@ def __init__(
         super().__init__(work_dir=work_dir, config=config)
         self.registry = MMPretrainRegistry()
 
+    def _update_evaluator(self, evaluator: list | dict | None, num_classes: int) -> list | dict | None:
+        if evaluator is None or not evaluator:
+            evaluator = [{"type": "Accuracy"}]
+        if isinstance(evaluator, list):
+            for metric in evaluator:
+                if isinstance(metric, dict):
+                    metric["_scope_"] = self.registry.name
+                    if "topk" in metric:
+                        metric["topk"] = [1] if num_classes < 5 else [1, 5]
+        elif isinstance(evaluator, dict):
+            evaluator["_scope_"] = self.registry.name
+            if "topk" in evaluator:
+                evaluator["topk"] = [1] if num_classes < 5 else [1, 5]
+        return evaluator
```

Comment:
It's not actual evaluator but evaluation config, right?
Not that important, but `_update_eval_config()` might be better to understand. Hope it could be refined at some point :)

Reply:
Yes, I agree. Thanks for the good suggestion. I'll fix it.